sklearn/metrics/_pairwise_distances_reduction.pyx

```diff
@@ -243,27 +283,13 @@ cdef class PairwiseDistancesReduction:
         the samples of Y selected by the reduction function.
         """
 
-        if strategy is None:
-            strategy = get_config().get("pairwise_dist_parallel_strategy", 'auto')
-
-        if strategy == 'auto':
-            # This is a simple heuristic whose constant for the
-            # comparison has been chosen based on experiments.
-            if 4 * self.chunk_size * self.effective_omp_n_thread < self.n_samples_X:
-                strategy = 'parallel_on_X'
-            else:
-                strategy = 'parallel_on_Y'
-
         # Limit the number of threads in second level of nested parallelism for BLAS
         # to avoid threads over-subscription (in GEMM for instance).
         with threadpool_limits(limits=1, user_api="blas"):
-            if strategy == 'parallel_on_Y':
+            if self.execute_in_parallel_on_Y:
                 self._parallel_on_Y()
-            elif strategy == 'parallel_on_X':
-                self._parallel_on_X()
             else:
-                raise RuntimeError(f"strategy must be 'parallel_on_X, 'parallel_on_Y', "
-                                   f"or 'auto', but currently strategy='{strategy}'.")
+                self._parallel_on_X()
 
         return self._finalize_results(return_distance)
```

Comment:
At this point `_compute` does not really do much anymore. Does it make sense to move this into `__init__`? 
Then `PairwiseDistancesArgKmin.compute` will look like:

```python
pda = PairwiseDistancesArgKmin(..., return_distance=True)
return pda.get_results()
```

Reply:
Definitely.